sdks/java/core/src/main/java/org/apache/beam/sdk/util/PubsubClient.java

```diff
@@ -54,6 +54,12 @@ PubsubClient newClient(
         @Nullable String timestampLabel,
         @Nullable String idLabel,
         PubsubOptions options) throws IOException;
+
+    /**
+     * Return the display name for this factory.
+     */
+    @Override
+    String toString();
```

Comment:
this seems an unorthodox re-use of Java Object.toString()

Not going to make you revert, but are you sure you want to do this?